lib/storage/src/content_manager/toc/transfer.rs

```diff
@@ -0,0 +1,28 @@
+use std::sync::Weak;
+
+use collection::shards::transfer::ShardTransferConsensus;
+
+use super::TableOfContent;
+use crate::content_manager::consensus_manager::ConsensusStateRef;
+
+#[derive(Clone)]
+pub struct ShardTransferDispatcher {
+    _toc: Weak<TableOfContent>,
+    consensus_state: ConsensusStateRef,
+}
```

Comment:
@ffuugoo We now have a custom type, holding a `Weak<T>` reference to prevent an `Arc` cycle.

One good thing is that this also removes the option from the fields.

It is currently still called *dispatcher, but we can decide to name it differently.